recipes/yaclib/all/conanfile.py

```diff
@@ -0,0 +1,159 @@
+from conan import ConanFile
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeToolchain, CMakeDeps
+from conan.tools.scm import Version
+from conan.tools.files import copy, patch, get
+from conan.errors import ConanInvalidConfiguration
+from functools import lru_cache
+import os
+
+required_conan_version = ">=1.47.0"
+
+
+class YACLibConan(ConanFile):
+    name = 'yaclib'
+    description = 'lightweight C++ library for concurrent and parallel task execution'
+    url = 'https://github.com/conan-io/conan-center-index'
+    homepage = 'https://github.com/YACLib/YACLib'
+    license = 'MIT'
+    topics = 'c++', 'async', 'parallel', 'concurrency'
+    settings = 'os', 'arch', 'compiler', 'build_type'
+    exports_sources = 'patches/**'
+
+    _yaclib_flags = {
+        'warn': [True, False],
+        'asan': [True, False],
+        'tsan': [True, False],
+        'ubsan': [True, False],
+        'lsan': [True, False],
+        'memsan': [True, False],
+        'coverage': [True, False],
+        'coro': [True, False],
+        'disable_futex': [True, False],
+        'disable_unsafe_futex': [True, False],
+        'disable_symmetric_transfer': [True, False],
+        'disable_final_suspend_transfer': [True, False],
+    }
+
+    _yaclib_faults = {
+        'thread_fault': [True, False],
+        'fiber_fault': [True, False],
+    }
+
+    options = {
+        'fPIC': [True, False],
+        **_yaclib_flags,
+        **_yaclib_faults,
+    }
+
+    default_options = {
+        'fPIC': True,
+        **{k: False for k in _yaclib_flags.keys()},
+        **{k: False for k in _yaclib_faults.keys()},
+    }
+
+    _cmake_definitions = {}
+
+    @property
+    def _source_subfolder(self):
```

Comment:
The problem is you are mixing the old and new styles in one recipe.

100 people same me make this milive 🙃 you need to delethis in favor of the `cmake_layout` to properly setup folders

Reply:
Now I feel that I should've spend more time reading the docs and examples.

Please forgive me :(

Reply:
Used layouts